File: mac/claude/claude-master-suite/scripts/python/indexnow_pusher.py
import sys
import json
import urllib.request
import urllib.error

def main():
    if len(sys.argv) < 4:
        print("Kullanım: python indexnow_pusher.py <host_adresi> <indexnow_key> <url_listesi_txt_yolu>")
        sys.exit(1)

    host = sys.argv[1]
    key = sys.argv[2]
    url_list_path = sys.argv[3]
    
    try:
        with open(url_list_path, 'r', encoding='utf-8') as f:
            urls = [line.strip() for line in f if line.strip()]
    except Exception as e:
        print(f"[X] HATA: Dosya okunamadı - {e}")
        sys.exit(1)

    data = {
        "host": host,
        "key": key,
        "keyLocation": f"https://{host}/{key}.txt",
        "urlList": urls
    }
    
    payload = json.dumps(data).encode('utf-8')
    req = urllib.request.Request(
        "https://api.indexnow.org/indexnow",
        data=payload,
        headers={'Content-Type': 'application/json; charset=utf-8'}
    )

    print("\n=== SÜPER SEO MOTORU (INDEXNOW PUSHER V3.0) ===")
    try:
        with urllib.request.urlopen(req) as response:
            status = response.getcode()
            print(f"[+] API Yanıt Kodu: {status}")
            if status in [200, 202]:
                print(f"[+] BAŞARILI: {len(urls)} URL arama motorlarına iletildi.")
            else:
                print(f"[X] HATA: Gönderim başarısız.")
    except urllib.error.URLError as e:
        print(f"[X] BAĞLANTI HATASI: {e}")

if __name__ == "__main__":
    main()

File: config/plugins/gemini-master-suite/skills/ag-seo-expert/scripts/js/indexnow-pusher.js
const https = require('https');

const host = process.argv[2];
const key = process.argv[3];
const urlListPath = process.argv[4];

if (!host || !key || !urlListPath) {
    console.log("Kullanım: node indexnow-pusher.js <host_adresi> <indexnow_key> <url_listesi_txt_yolu>");
    process.exit(1);
}

const fs = require('fs');
const urls = fs.readFileSync(urlListPath, 'utf8').split('\n').map(l => l.trim()).filter(l => l.length > 0);

const payload = JSON.stringify({
    host: host,
    key: key,
    keyLocation: `https://${host}/${key}.txt`,
    urlList: urls
});

const options = {
    hostname: 'api.indexnow.org',
    path: '/indexnow',
    method: 'POST',
    headers: {
        'Content-Type': 'application/json',
        'Content-Length': Buffer.byteLength(payload)
    }
};

const req = https.request(options, (res) => {
    console.log(`\n=== SÜPER SEO MOTORU (INDEXNOW PUSHER V3.0) ===`);
    console.log(`[+] API Yanıt Kodu: ${res.statusCode}`);
    if (res.statusCode === 200 || res.statusCode === 202) {
        console.log(`[+] BAŞARILI: ${urls.length} URL arama motorlarına iletildi.`);
    } else {
        console.log(`[X] HATA: Gönderim başarısız.`);
    }
});

req.on('error', (e) => {
    console.error(`[X] BAĞLANTI HATASI: ${e.message}`);
});

req.write(payload);
req.end();
